### DISCUSSION TITLE
Works fine also on 433Mhz remote

### DISCUSSION THREAD

**Discussion by Totoche95** [bengtmartensson/IrScrutinizer#504] [Ideas]
This program works really fine and memorize and analyse all my IR Remote.
I put 433Mhz receiver module on br 5 Arduino , inverting phase with a transistor (you have to put "use receive for capture" in hardware, and the 433Mhz emitter module input on br 3 Arduino.
I can memorise my 433Mhz alarm remote et activate or desativate alarm with IRScrutinizer, works fine.
Two little things to improve the system.
I think there is a 38 Khz carrier added, if it is possible to add an option to desactivate carrier, would be fine
If it is possible add an option to invert phase on input/output, this would be fine for testing other use of IRScrutinizer.
I am desapointed, i thinked my alarm working with rolling codes, but nothing like that. I can use the code memorized with IRScrutinier 10 times, it works each time.
Cheers and thanks to devellopers (i was IT and occasionaly develloper)

**Comment by bengtmartensson** [comment]
Thank you for those kind words. 

I have been using a simple module to send 433 MHz signals (in particular with the Arctech, which IrScrutinizer handles). 

> I think there is a 38 Khz carrier added, if it is possible to add an option to desactivate carrier, would be fine

I am not quite sure what you mean. If you define `NON_MOD` (see `GirsFat_config.h` in AGirs),  define `NON_MOD_PIN` to the output pin you want to use, signals with modulation frequency 0 (like Arctech) will be non-modulatedly output on `NON_MOD_PIN`.

However, I have not had very much success with receiving such signals -- but I did not try very hard either. Appears that the AGC (automatic gain control) turns up the gain between the sends, which makes it pick up a lot of noise. If you have had success with this, I would be happy if you describe it in detail.

> If it is possible add an option to invert phase on input/output, this would be fine for testing other use of IRScrutinizer.

Sending: The sending class is `IrSenderNonMod`of Infrared4Arduino. Its constructor has a second argument `invert`, defaulted to `false`. Calling with `true` will invert the sending signal (line 170 in Girs.cpp).
Receiving: change `invertingSensor` in IrReceiver to false (Line 23 in IrReceiver.h).


> I am desapointed, i thinked my alarm working with rolling codes, but nothing like that. I can use the code memorized with IRScrutinier 10 times, it works each time.

I'd say that you found out in a fairly painless way...

**Comment by Totoche95** [comment]
Thanks for your reply.
For the carrier option, i added in GirsFat.config.h and GirsLite.config.h (i have Arduino UNO) : #define NON_MOD true, #define NON_MOD_PIN 4 and  pinMode(NON_MOD_PIN,OUTPUT) in Girs.cpp but nothing on output 4 of arduino when i click Transmit, something i did not understand, 3 of arduino continue to output good signal.(Same result if i take pin12)
signals with modulation frequency 0 (like Arctech) will be non-modulatedly output on NON_MOD_PIN
Do i have also to put frequency to 0 somewhere ?

I use also a very cheap 433Mhz receiver and the solution i found is to press the remote button before capturing, my receiver takes also to much noise, by this way it works. I have also to invert signal between receiver and Arduino entry (br5) with one transistor. (Can also invert on the emitter command but in this case the emitter will be ON all the time, but it transmit also correctly.
I will try this heterodyne receiver to let IRScrutinizer in capture :
[](url)https://ph-test-11.slatic.net/p/b9e8d62230ef625f1ebaab98b5622aa1.jpg
Can find on Aliexpress for 2 $.(3 V -3V3 power supply)
[](url)https://fr.aliexpress.com/item/1005003436580019.html?gatewayAdapt=glo2fra
I found discussion here, konbaasiang reply is interresting, as you said it it AGC control on 0.5 $ modules
[](url)https://www.reddit.com/r/arduino/comments/xvncbx/i_cant_get_the_433mhz_rf_receiver_to_work_it_just/?rdt=60859
Reception schematic is in attachment                
[Decodeur-433Mhz.pdf](https://github.com/bengtmartensson/IrScrutinizer/files/13249739/Decodeur-433Mhz.pdf)

**Reply by bengtmartensson** [reply]
Thanx for the links, interesting. Seems to agree with my quick guesses.

> nothing on output 4 of arduino when i click Transmit, something i did not understand, 3 of arduino continue to > output good signal.(Same result if i take pin12)
> signals with modulation frequency 0 (like Arctech) will be non-modulatedly output on NON_MOD_PIN
> Do i have also to put frequency to 0 somewhere ?

The way Girs.cpp/sendIrSignal is written, it sends signals with frequency 0 to NON_MOD_PIN, and others to the modulating sender: See lines 168-174:


```
    IrSender *irSender =
#ifdef NON_MOD
            (irSignal.getFrequency() == 0) ? (IrSender*) new IrSenderNonMod(NON_MOD_PIN) :
#endif
            (IrSender*) IrSenderPwm::getInstance(true);

    irSender->sendIrSignal(irSignal, noSends);
```

Feel free to modify it. But a non-zero modulation frequency does not really make sense on 433 MHz signals.

**Comment by Totoche95** [comment]
Thanks for your reply.
As i did'nt found how to put frequency to 0 from IRScrutinizer i put a little switch on pin 2 (carrier_switch) on my Arduino shield , 
#define NON_MOD_PIN 10 and modified lines :

IrSender *irSender =
#ifdef NON_MOD
            (carrier_switch == LOW) ? (IrSender*) new IrSenderNonMod(NON_MOD_PIN) :
#endif
            (IrSender*) IrSenderPwm::getInstance(true);

    irSender->sendIrSignal(irSignal, noSends);
    
    It works fine like that output on pin 3 for IR, pin 10 for radio
    Thanks for your help

**Comment by bengtmartensson** [comment]
If you are using it only for a limited use case (for example no modulated sends), feel free to configure/rewrite for that usecase. (I see that NON_MOD without TRANSMIT will not work, will try to fix for the next release.)

**Comment by Totoche95** [comment]
Thanks, for me, it works fine like that, i have a little switch on my arduino shield : carrier / no carrier . Signal with carrier on pin 3, without carrier on pin 10.
Thanks for your help
